test/python/quantum_info/states/test_statevector.py

```diff
@@ -41,6 +41,61 @@
 class TestStatevector(QiskitTestCase):
     """Tests for Statevector class."""
 
+    def test_from_circuit_bell_state(self):
+        """Test from_circuit creates Bell state correctly."""
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0, 1)
+
+        sv = Statevector.from_circuit(qc)
+        expected_data = np.array([1 / np.sqrt(2), 0, 0, 1 / np.sqrt(2)])
+        expected = Statevector(expected_data)
+
+        self.assertTrue(sv.equiv(expected))
+
+    def test_from_circuit_transpilation_consistency(self):
+        """Test transpilation consistency - the main GitHub issue."""
+        qc = QuantumCircuit(3)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.swap(1, 2)
+        qc.z(0)
+
+        backend = BasicSimulator()
+        transpiled_opt0 = transpile(qc, backend, optimization_level=0)
+        transpiled_opt2 = transpile(qc, backend, optimization_level=2)
+
+        sv1 = Statevector.from_circuit(transpiled_opt0)
+        sv2 = Statevector.from_circuit(transpiled_opt2)
+
+        self.assertTrue(sv1.equiv(sv2))
```

Comment:
What's not tested yet is that `ignore_set_layout=True` actually does ignore the layout _if one exists_. Could you add this case to check here? E.g.
```suggestion
        self.assertTrue(sv1.equiv(sv2))
        
        sv2_nolayout = Statevector.from_circuit(transpiled_opt2, ignore_set_layout=True)
        self.assertFalse(sv1.equiv(sv2_nolayout))
```